src/pages/_app.page.tsx

```diff
@@ -36,9 +36,10 @@ const AppWrapper: React.FC = ({ children }) => {
   ]
 
   useEffect(() => {
-    if (!signerNetwork) return
+    if (!signerNetwork || !userAddress) return
```

Comment:
If the user isn't connected, bail early.

When the user is connected AND also on the wrong network, then we'll still show the annoying modal.